### PR TITLE
Adds Cypress & Sonar

### PR DESCRIPTION
**Pull request by sam-drumm**
https://hackney.atlassian.net/browse/TS-1580?atlOrigin=eyJpIjoiOTA5NTUwNmI5YjNiNDcxZWEwZTViNDc3YjRiODJjMTMiLCJwIjoiaiJ9

**What**
- Adds configuration and packages for cypress e2e tests
- Adds configuration to run Sonar in the pipeline

**Why** 
- We want to expand testing coverage to include some e2e tests
- We want sonar to check our code 

**How**
- Adds the required libraries and config to run cypress
- Updates the CI config to run cypress test on merge into development and make passing a requirement to build stage
- Updates CI config to run sonar in parallel with test step

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+type Environment = 'localdev' | 'development' | 'staging';
+
+const environment: Environment =
+  (process.env.NEXT_PUBLIC_ENV as Environment) || 'development';
+
+const baseUrlSites: Record<Environment, string> = {
+  localdev: 'http://localhost:3000',
+  development: 'https://housing-register-development.hackney.gov.uk/',
+  staging: 'https://housing-register-staging.hackney.gov.uk/',
```
Comment (LBHTKarki) [inline cypress.config.ts:12]: We probably don't need staging unless we plan on running against it too.

Comment (sam-drumm) [reply cypress.config.ts:12]: Cool updated.

File: cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  localdev: 'http://localhost:3000',
+  development: 'https://housing-register-development.hackney.gov.uk/',
+};
+
+const baseUrl = baseUrlSites[environment];
+
+export default defineConfig({
+  e2e: {
+    baseUrl,
+    blockHosts: ['*.google-analytics.com', '*.sentry.io'],
```
Comment (LBHTKarki) [inline cypress.config.ts:19]: This can probably be removed now that sentry and analytics have been handled elsewhere

Comment (sam-drumm) [reply cypress.config.ts]: Cleaned up now.

**Review by LBHTKarki — APPROVED** [review]
Looks good👍 We can ignore the Sonar results for these initial setup PRs

**Pull request merged**